homework_2/cmd/bootstrap/axilisOfferFeed2.go

```diff
@@ -0,0 +1,12 @@
+package bootstrap
+
+import (
+	stdhttp "net/http"
+
+	"code-cadets-2021/lecture_2/06_offerfeed/internal/infrastructure/http"
+)
+
+func AxilisOfferFeed2() *http.AxilisOfferFeed2 {
+	httpClient := &stdhttp.Client{}
```

Comment:
Just one more detail. (Actually not a scope of this homework, but lets improve it even more).

Whole application should use 1 instance of `http.Client`. You have two feeds and each of them is creating its own client.

In `bootstrap`, you can create a method which will bootstrap an http client for you. Then you can pass the same client to both feed constructors.

Also, when you are interacting with dependencies such as DBs, HTTP servers etc. (network operations) always use timeouts.
So create `http.Client` like this:
```
	http.Client{
		Timeout: time.Second * 10,
	}
```

---

Why single client in the app:
Single client should create all TCP connections in the app, it can reuse them then. If there are two clients, those could open too many connections (they dont know about each other) and that could crash the app.
Go documentation also states that single client should be created in each app and reused.

Reply:
I changed it like this :)